src/distance/JointAvoidanceConfigurationRanker.cpp

```diff
@@ -0,0 +1,59 @@
+#include "aikido/distance/JointAvoidanceConfigurationRanker.hpp"
+
+namespace aikido {
+namespace distance {
+
+using statespace::dart::ConstMetaSkeletonStateSpacePtr;
+using ::dart::dynamics::ConstMetaSkeletonPtr;
+
+//==============================================================================
+JointAvoidanceConfigurationRanker::JointAvoidanceConfigurationRanker(
+    ConstMetaSkeletonStateSpacePtr metaSkeletonStateSpace,
+    ConstMetaSkeletonPtr metaSkeleton)
+  : ConfigurationRanker(
+        std::move(metaSkeletonStateSpace), std::move(metaSkeleton))
+{
+  auto lowerLimits = mMetaSkeleton->getPositionLowerLimits();
+  auto upperLimits = mMetaSkeleton->getPositionUpperLimits();
+
+  for (std::size_t i = 0; i < mMetaSkeletonStateSpace->getDimension(); ++i)
+  {
+    if (lowerLimits[i] == -dart::math::constantsd::inf())
+      mUnboundedLowerLimitsIndices.emplace_back(i);
+
+    if (upperLimits[i] == dart::math::constantsd::inf())
+      mUnboundedUpperLimitsIndices.emplace_back(i);
+  }
+
+  mLowerLimitsState = mMetaSkeletonStateSpace->createState();
+  mUpperLimitsState = mMetaSkeletonStateSpace->createState();
```

Comment:
`mUpperLimitsState` should be scoped state. Otherwise it will be destructed out of this function, and `mUpperLimitsState` will become dangling pointer.